File: hacker_rank/data structures/arrays/01_array_ds.py
import sys

"""
Given an array, , of  integers, print each element in reverse order as a single line of space-separated integers.

Sample Input
4
1 4 3 2

Sample Output
2 3 4 1
"""
n = int(input().strip())
arr = [int(arr_temp) for arr_temp in input().strip().split(' ')]

if n != len(arr):
    print("Array out of range")
else:
    for i in range(0,int(len(arr)/2)):
        temp = arr[i]
        arr[i] = arr[len(arr) - i - 1]
        arr[len(arr) - i - 1] = temp
for i in arr:
    print(i, end=" ")

File: hacker_rank/data structures/arrays/01_array_ds.cpp
// Given an array, , of  integers, print each element in reverse order as a single line of space-separated integers.

// Sample Input
// 4
// 1 4 3 2

// Sample Output
// 2 3 4 1

// By Deepanshu Tyagi on 9th July 14:02

#include <iostream>
#include <string>
#include <vector>

using namespace std;

class ArrayDs{
    public:
    void reverseArray(vector<int> array){
        
        for(int i = array.size() - 1; i >= 0; i--)
        {
            cout << array[i] << " ";
        }
    }

    public:
    void main(){
        int len;
        cin >> len;
        vector<int> array;
        for(int i = 0; i < len; i++)
        {
            int value;
            cin >> value;
            array.push_back(value);
        }
        reverseArray(array);
    }
};

int main()
{
    ArrayDs arrayDs;
    arrayDs.main();
    return 0;
}
